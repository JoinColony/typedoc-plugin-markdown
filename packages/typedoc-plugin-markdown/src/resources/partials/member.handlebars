{{#unless hasOwnDocument}}

{{#if name}}

### {{#if @options.namedAnchors}}<a id="{{anchor}}" name="{{this.anchor}}"></a> {{/if}}{{ escapeChars name }}

{{/if}}

{{/unless}}

{{#if signatures}}

{{#each signatures}}

{{> member.signature showSources=true }}

{{/each}}

{{else}}

{{#if hasGetterOrSetter}}

{{> member.getterSetter}}

{{else}}

{{#ifIsReference this}}

{{> member.reference  }}

{{else}}

{{> member.declaration}}

{{/ifIsReference}}

{{/if}}

{{/if}}

{{#unless @last}}
{{#unless hasOwnDocument}}
___
{{/unless}}
{{/unless}}
