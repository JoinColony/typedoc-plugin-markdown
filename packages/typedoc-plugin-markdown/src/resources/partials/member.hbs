{{#if signatures}}

{{#each signatures}}

### {{> member.signature showSources=true }}

{{/each}}

{{else}}

{{#unless hasOwnDocument}}

{{#if name}}

### {{#ifNamedAnchors}}<a id="{{anchor}}" name="{{this.anchor}}"></a> {{/ifNamedAnchors}}{{ escape name }}

{{/if}}

{{/unless}}

{{#if hasGetterOrSetter}}

{{> member.getterSetter}}

{{else}}

{{#ifIsReference}}

{{referenceMember}}

{{else}}

{{> member.declaration}}

{{/ifIsReference}}

{{/if}}

{{/if}}
