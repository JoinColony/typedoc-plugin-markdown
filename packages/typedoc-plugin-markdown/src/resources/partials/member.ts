import { DeclarationReflection, ReferenceReflection } from 'typedoc';
import { MarkdownThemeContext } from '../..';
import { escapeChars } from '../../utils/format';
import { heading } from '../../utils/elements';

export const memberPartial = (
  context: MarkdownThemeContext,
  props: DeclarationReflection | ReferenceReflection,
) => {
  const md: string[] = [];
  md.push(
    heading(
      3,
      context.options.namedAnchors
        ? `<a id="${props.anchor}" name="${props.anchor}"></a> `
        : '' + escapeChars(props.name),
    ),
  );
  if (props.signatures) {
    props.signatures.forEach((signature) => {
      md.push(context.signatureMemberPartial(signature));
    });
  } else {
    if (props.hasGetterOrSetter()) {
      if (props.getSignature) {
        md.push(
          context.signatureMemberPartial(props.getSignature, false, 'get'),
        );
      }
      if (props.setSignature) {
        md.push(
          context.signatureMemberPartial(props.setSignature, false, 'set'),
        );
      }
    } else {
      if (props instanceof ReferenceReflection) {
        md.push(context.referenceMemberPartial(props));
      }
      if (props instanceof DeclarationReflection) {
        md.push(context.declarationMemberPartial(props));
      }
    }
  }
  return md.join('\n\n');
};
