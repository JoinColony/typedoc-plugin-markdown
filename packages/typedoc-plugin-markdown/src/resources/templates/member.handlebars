{{> header}}

{{#with model}}

{{> member}}

{{/with}}
