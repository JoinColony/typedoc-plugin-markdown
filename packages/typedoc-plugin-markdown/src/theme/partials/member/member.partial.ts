import { DeclarationReflection, ReferenceReflection } from 'typedoc';
import { MarkdownThemeRenderContext } from '../../theme.context';
import { escapeChars, heading } from '../../theme.utils';

export const memberPartial = (
  context: MarkdownThemeRenderContext,
  props: DeclarationReflection | ReferenceReflection,
) => {
  const md: string[] = [];

  if (context.options.namedAnchors) {
    md.push(`<a id="${props.anchor}" name="${props.anchor}"></a>`);
  }

  md.push(heading(3, escapeChars(props.name)));

  if (props.signatures) {
    props.signatures.forEach((signature) => {
      md.push(context.signaturePartial(signature));
    });
  } else {
    if (props.hasGetterOrSetter()) {
      if (props.getSignature) {
        md.push(context.signaturePartial(props.getSignature, false, 'get'));
      }
      if (props.setSignature) {
        md.push(context.signaturePartial(props.setSignature, false, 'set'));
      }
    } else {
      if (props instanceof ReferenceReflection) {
        md.push(context.referencePartial(props));
      } else if (props instanceof DeclarationReflection) {
        md.push(context.declarationPartial(props));
      }
    }
  }
  return md.join('\n\n');
};
