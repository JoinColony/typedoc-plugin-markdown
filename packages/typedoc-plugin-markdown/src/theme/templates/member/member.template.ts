import { DeclarationReflection, PageEvent } from 'typedoc';
import { MarkdownThemeRenderContext } from '../../theme.context';
import { heading } from '../../theme.utils';

export const memberTemplate = (
  context: MarkdownThemeRenderContext,
  props: PageEvent<DeclarationReflection>,
) => {
  const md: string[] = [];

  if (!context.getOption('hideBreadcrumbs')) {
    md.push(context.breadcrumbsPartial(props));
  }

  if (!context.getOption('hidePageTitle')) {
    md.push(heading(1, context.pageTitlePartial(props, true)));
  }

  if (props.model) {
    md.push(context.memberPartial(props.model));
  }

  return md.join('\n\n');
};
