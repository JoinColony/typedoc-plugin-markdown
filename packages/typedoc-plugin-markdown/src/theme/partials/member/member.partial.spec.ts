import { DeclarationReflection, ProjectReflection } from 'typedoc';
import { MarkdownThemeRenderContext } from '../..';
import { memberPartial } from './member.partial';

describe(`Member Partial:`, () => {
  let project: ProjectReflection;
  let context: MarkdownThemeRenderContext;

  beforeAll(async () => {
    project = global.getProject('members.ts');
  });

  test(`should compile declaration member'`, () => {
    expect(
      memberPartial(
        global.getMockContext(),
        project.findReflectionByName(
          'declarationMember',
        ) as DeclarationReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile a signature member'`, () => {
    expect(
      memberPartial(
        global.getMockContext(),
        project.findReflectionByName(
          'signatureMember',
        ) as DeclarationReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile a referenced member'`, () => {
    expect(
      memberPartial(
        global.getMockContext(),
        project.findReflectionByName(
          'ReferencedMember',
        ) as DeclarationReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile members with getter'`, () => {
    expect(
      memberPartial(
        global.getMockContext(),
        (
          project.findReflectionByName(
            'ClassWithAccessorMembers',
          ) as DeclarationReflection
        ).findReflectionByName('getter') as DeclarationReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile members with setter'`, () => {
    expect(
      memberPartial(
        global.getMockContext(),
        (
          project.findReflectionByName(
            'ClassWithAccessorMembers',
          ) as DeclarationReflection
        ).findReflectionByName('setter') as DeclarationReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should set named anchors'`, () => {
    expect(
      memberPartial(
        global.getMockContext({ namedAnchors: true }),
        project.findReflectionByName(
          'declarationMember',
        ) as DeclarationReflection,
      ),
    ).toMatchSnapshot();
  });
});
