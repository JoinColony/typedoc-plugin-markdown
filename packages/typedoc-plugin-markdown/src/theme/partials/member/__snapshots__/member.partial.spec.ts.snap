// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member Partial: should compile a referenced member' 1`] = `
"### ReferencedMember

{ referencePartial }"
`;

exports[`Member Partial: should compile a signature member' 1`] = `
"### signatureMember

{ reflectionTitlePartial }"
`;

exports[`Member Partial: should compile declaration member' 1`] = `
"### declarationMember

{ declaration }"
`;

exports[`Member Partial: should compile members with getter' 1`] = `
"### getter

{ reflectionTitlePartial }"
`;

exports[`Member Partial: should compile members with setter' 1`] = `
"### setter

{ reflectionTitlePartial }"
`;

exports[`Member Partial: should set named anchors' 1`] = `
"<a id=\\"undefined\\" name=\\"undefined\\"></a>

### declarationMember

{ declaration }"
`;
